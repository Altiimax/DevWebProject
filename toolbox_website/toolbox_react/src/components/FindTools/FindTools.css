.baseForm {
  background-color: white;
  padding: 10px;
  border-radius: 0px 0px 10px 10px;
}

.searchForm {
  color: black;
  background-color: rgb(255, 255, 255);
  border-radius: 10px;
  padding: 20px;
  margin: 1em;
}

.formTitle {
  border-radius: 10px 10px 0px 0px;
  color: black;
  margin-bottom: 0;
  text-align: center;
  padding: 10px;
}

.searchButton {
  color: white;
  background-color: rgb(192, 11, 11);
  border: none;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  border-radius: 10px;
  display: flex;
  justify-content: center;
  font-size: 130%;
  padding: 0.3em 0em;
}

.searchButton:hover {
  color: white;
  background-color: red;
  box-shadow: 5px 5px 10px black;
  border: solid 1px red;
}

.noResults{
  margin: 1em 1em;
  font-size: larger;
}