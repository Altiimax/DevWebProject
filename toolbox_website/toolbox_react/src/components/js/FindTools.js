import React, { Component } from "react";
import { Form, Col } from "react-bootstrap";
import "../css/FindTools.css";

class FindTools extends Component {
  render() {
    return (
      <div>
        <h1 className="formTitle">Find nearby tools</h1>
        <Form className="baseForm">
          <Form.Group>
            <Form.Label>Which tool?</Form.Label>
            <Form.Control
              type="text"
              placeholder="Wrench, saw, screwdriver, ..."
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>Where?</Form.Label>
            <Form.Control type="text" placeholder="Select a city by name" />
          </Form.Group>
          <Form.Row>
            <Form.Group as={Col}>
              <Form.Label>From</Form.Label>
              <Form.Control type="date" />
            </Form.Group>
            <Form.Group as={Col}>
              <Form.Label>To</Form.Label>
              <Form.Control type="date" />
            </Form.Group>
          </Form.Row>
          <button className="searchButton" type="submit">
            Research
          </button>
        </Form>
      </div>
    );
  }
}

export default FindTools;
