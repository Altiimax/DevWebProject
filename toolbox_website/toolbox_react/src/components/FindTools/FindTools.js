import React, { Component } from "react";
import { Form /*Col*/ } from "react-bootstrap";
import "./FindTools.css";
import { apiRequest } from "../../api/apiRequest.js";
import Search from "../Search/Search.js";
import ReactDOM from "react-dom";

class FindTools extends Component {
  constructor(props) {
    super(props);
    this.state = {
      toolName: "",
      townName: "",
      validated: false,
    };
  }

  apiFindTools = () => {
    let self = this;
    let endpoint =
      "/api/search/?what='" +
      self.state.toolName +
      "'&where='" +
      self.state.townName +
      "'";

    let req = new apiRequest();
    req.open("GET", `${endpoint}`);

    req.addEventListener("readystatechange", function () {
      if (this.readyState === 4) {
        if (this.status === 200) {
          let rep = JSON.parse(this.responseText);
          console.log(rep);
          if (rep.length === 0) {
            console.log("Tool not found");
          } else {
            ReactDOM.render(
              <Search data={rep} />,
              document.getElementById("list")
            );
          }
        }
      }
    });

    req.send();
    console.log(this.state.toolName);
    console.log(this.state.townName);
  };

  handleChange = (e) => {
    let target = e.target;
    let value = target.value;
    let name = target.name;
    this.setState({ [name]: value });
  };

  handleSubmit = (e) => {
    const form = e.currentTarget;
    if(form.checkValidity() === false){
        e.preventDefault();
        e.stopPropagation();
    }
    this.setState({validated : true});
        e.preventDefault();
        this.apiFindTools();
      };

  render() {
    return (
      <div className="searchForm">
        <h1 className="formTitle">Find nearby tools</h1>
        <Form className="baseForm" noValidate validated={this.state.validated}>
          <Form.Group>
            <Form.Label>Which tool?</Form.Label>
            <Form.Control
              required
              type="text"
              name="toolName"
              placeholder="Wrench, saw, screwdriver, ..."
              onChange={this.handleChange}
            />
            <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
            <Form.Control.Feedback type="invalid">This field is required!</Form.Control.Feedback>
          </Form.Group>
          <Form.Group>
            <Form.Label>Where?</Form.Label>
            <Form.Control
              required
              type="text"
              placeholder="Select a city by name"
              name="townName"
              onChange={this.handleChange}
            />
            <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
            <Form.Control.Feedback type="invalid">This field is required!</Form.Control.Feedback>
          </Form.Group>
          {/* //TODO pour le moment on implémente pas cette fonctionalitée ;) 
          <Form.Row>
            <Form.Group as={Col}>
              <Form.Label>From</Form.Label>
              <Form.Control type="date" />
            </Form.Group>
            <Form.Group as={Col}>
              <Form.Label>To</Form.Label>
              <Form.Control type="date" />
            </Form.Group>
          </Form.Row>*/}
          <div className="searchButtonContainer">
            <button
              className="searchButton"
              type="submit"
              onClick={this.handleSubmit}
            >
              Search
            </button>
          </div>
        </Form>
      </div>
    );
  }
}

export default FindTools;
