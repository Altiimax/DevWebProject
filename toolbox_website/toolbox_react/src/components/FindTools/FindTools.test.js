import React from "react";
import ReactDOM from "react-dom";
import FindTools from "./FindTools.js";
it("render without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<FindTools />, div);
});
