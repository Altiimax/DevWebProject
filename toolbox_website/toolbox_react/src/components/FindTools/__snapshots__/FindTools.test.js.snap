// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form 1`] = `
<div
  className="searchForm"
>
  <h1
    className="formTitle"
  >
    Find nearby tools
  </h1>
  <form
    className="baseForm"
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Which tool?
      </label>
      <input
        className="form-control"
        placeholder="Wrench, saw, screwdriver, ..."
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Where?
      </label>
      <input
        className="form-control"
        placeholder="Select a city by name"
        type="text"
      />
    </div>
    <div
      className="searchButtonContainer"
    >
      <button
        className="searchButton"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
